2.5/README.md

```diff
@@ -23,26 +23,111 @@ modules for their web applications. There is no guarantee for any specific npm o
-    ```
+Source-to-Image framework and scripts
+-------------------------------------
+This image supports the [Source-to-Image](https://docs.openshift.com/container-platform/3.11/creating_images/s2i.html)
+(S2I) strategy in OpenShift. The Source-to-Image is an OpenShift framework
+which makes it easy to write images that take application source code as
+an input, use a builder image like this Node.js container image, and produce
+a new image that runs the assembled application as an output.
 
-**Accessing the application:**
+To support the Source-to-Image framework, important scripts are included in the builder image:
+
+* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts. The script takes sources of a given application and places them into appropriate directories inside the image. It utilizes some common patterns in Ruby application development (see the **Environment variables** section below).
+* The `/usr/libexec/s2i/run` script is set as the default command in the resulting container image (the new image with the application artifacts). It runs `rackup`.
+
+Building an application using a Dockerfile
+---------------------
+Compared to the Source-to-Image strategy, using a Dockerfile is a more
+flexible way to build a Node.js container image with an application.
+Use a Dockerfile when Source-to-Image is not sufficiently flexible for you or
+when you build the image outside of the OpenShift environment.
+
+To use the Ruby image in a Dockerfile, follow these steps:
+
+#### 1. Pull a base builder image to build on
+
+```
+podman pull ubi8/ruby-25
+```
+
+An RHEL7 image `ubi8/ruby-25` is used in this example.
+
+#### 2. Pull and application code
+
+ An example application available at https://github.com/sclorg/rails-ex.git is used here. Feel free to clone the repository for further experiments.
+
+```
+git clone https://github.com/sclorg/rails-ex.git app-src
 ```
-$ curl 127.0.0.1:8080
+
+#### 3. Prepare an application inside a container
+
+This step usually consists of at least these parts:
+
+* putting the application source into the container
+* installing the dependencies
+* setting the default command in the resulting image
+
+For all these three parts, users can either setup all manually and use commands `ruby`, `bundle` and `rackup` explicitly in the Dockerfile ([3.1.](#31-to-use-your-own-setup-create-a-dockerfile-with-this-content)), or users can use the Source-to-Image scripts inside the image ([3.2.](#32-to-use-the-source-to-image-scripts-and-build-an-image-using-a-dockerfile-create-a-dockerfile-with-this-content); see more about these scripts in the section "Source-to-Image framework and scripts" above), that already know how to set-up and run some common Perl applications.
+
+##### 3.1 To use your own setup, create a Dockerfile with this content:
```

Comment:
Why would you recommend something like this, instead of `3.2`? If really needed, could we switch the order, so users read the recommended solution first?

Reply:
this order is the same in each container, @hhorak what is your oppinion should we change it only here or everywhere?

Reply:
I think starting with the simpler example is ok here.